app/controllers/base_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class BaseController
+  DIFFICULTIES = CodebreakerDiz::DIFFICULTIES
+  PATHS = ['./app/views/', './app/views/partials/'].freeze
+
+  private_constant :PATHS
+
+  def self.find_views
+    regex = Regexp.new('.+html.haml')
+
+    views = { common: {}, partial: {} }
+
+    PATHS.each_with_index do |path, index|
+      Dir.entries(path).each do |file|
+        views[views.keys[index]][file.split('.').first.intern] = file if file.match?(regex)
+      end
+    end
+
+    views
+  end
+
+  VIEWS ||= find_views
```

Comment:
move it to `views` method with memoization: https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/

Reply:
![image](https://user-images.githubusercontent.com/38437488/60272074-261b1380-98fc-11e9-9bc1-7bd80376fcfa.png)
Is that correct?